src/imitation/rewards/serialize.py

```diff
@@ -17,33 +23,65 @@
 reward_registry: registry.Registry[RewardFnLoaderFn] = registry.Registry()
 
 
-def _load_reward_net_as_fn(shaped: bool) -> RewardFnLoaderFn:
-    def loader(path: str, venv: VecEnv) -> common.RewardFn:
-        """Load train (shaped) or test (not shaped) reward from path."""
-        del venv  # Unused.
-        net = th.load(str(path))
-        if not shaped and hasattr(net, "base"):
-            # If the "base" attribute exists, we are dealing with a ShapedRewardNet
-            # and will disable the potential shaping (if shaped is False).
-            # If no "base" attribute exists, we seem to use an unshaped RewardNet
-            # anyway, so we just use its predict() method directly.
-            reward = net.base.predict
+def _validate_reward(reward: common.RewardFn) -> common.RewardFn:
+    """Add sanity check to the reward function for dealing with VecEnvs."""
+
+    def rew_fn(
+        obs: np.ndarray,
+        act: np.ndarray,
+        next_obs: np.ndarray,
+        dones: np.ndarray,
+    ) -> np.ndarray:
+        rew = reward(obs, act, next_obs, dones)
+        assert rew.shape == (len(obs),)
+        return rew
+
+    return rew_fn
+
+
+def _strip_wrappers(
+    net: RewardNet,
+    wrappers: List[Type[RewardNetWrapper]],
+) -> RewardNet:
+    """Attempts to remove provided wrappers.
+
+    wrappers until either it reaches a non-warper
+    reward net or it has removed all the listed wrappers.
+
+    Args:
+        net: an instance of a reward network that may be wrapped
+        wrappers: list of wrapper types to remove
+
+    Returns:
+        The reward network with the listed wrappers removed
+    """
+    for wrapper_type in wrappers:
+        if not isinstance(net, RewardNetWrapper):
+            # Reached base reward net
+            break
+
+        assert hasattr(net, "base")
+
+        if isinstance(net, wrapper_type):
+            net = net.base
         else:
-            reward = net.predict
+            break
 
-        def rew_fn(
-            obs: np.ndarray,
-            act: np.ndarray,
-            next_obs: np.ndarray,
-            dones: np.ndarray,
-        ) -> np.ndarray:
-            rew = reward(obs, act, next_obs, dones)
-            assert rew.shape == (len(obs),)
-            return rew
+    return net
 
-        return rew_fn
 
-    return loader
+def _make_functional(
+    net: RewardNet,
+    attr: str = "predict",
+    kwargs=dict(),
+) -> common.RewardFn:
+    return lambda *args: getattr(net, attr)(*args, **kwargs)
+
+
+def _validate_type(net: RewardNet, type_: Type[RewardNet]):
```

Comment:
Change the function name to sth like `_validate_norm_reward_type`? Because the check doesn't apply to `RewardNet_shaped`. Another alternative would be to also extend it to `RewardNet_shaped`?

Reply:
Ya I could definitely have it also validate that the reward function you are loading is shaped. I was just a bit nervous about breaking the current interface for anyone already using the code.

Reply:
It turns out adding validation did break some test cases. I think there is likely either some caused by some other bug in the code or these test cases where not correct. See: https://github.com/HumanCompatibleAI/imitation/pull/440#discussion_r885117806